os/migrating-to-clcs.md

```diff
@@ -0,0 +1,373 @@
+
+This document details how to convert an existing cloud config into a CLC. For information on what to do with a CLC, take a look at the [Config Transpiler's Getting Started page][ct-getting-started].
+
+The etcd and flannel examples shown in this document will use dynamic data in the CLC (anything looking like this: `{PRIVATE_IPV4}`). Not all types of dynamic data are supported on all cloud providers, and if the machine is not on a cloud provider this feature cannot be used. Please see [here][dynamic-data] for more information.
+
+To see all supported options available in a CLC, please look at the [CLC schema][ct-config].
+
+## coreos
+
+### etcd2
+
+In a cloud config, etcd version 2 can be enabled and configured by using the `coreos.etcd.*` section. As an example of this:
+
+```cloud-config
+#cloud-config
+
+coreos:
+  etcd2:
+    discovery:                   "https://discovery.etcd.io/<token>"
+    advertise-client-urls:       "http://$public_ipv4:2379"
+    initial-advertise-peer-urls: "http://$private_ipv4:2380"
+    listen-client-urls:          "http://0.0.0.0:2379,http://0.0.0.0:4001"
+    listen-peer-urls:            "http://$private_ipv4:2380,http://$private_ipv4:7001"
+```
+
+In a CLC etcd can also be configured, but a CLC will use the etcd-member service on a Container Linux machine instead of the etcd2 service like cloudinit would.  The etcd-member service will download a version of etcd of the user's choosing and run it. This means that in a CLC both etcd v2 and v3 can be configured.
+
+This is done under the etcd section:
+
+```container-linux-config
+etcd:
+    version: 3.1.6
+```
+
+The version can also be omitted to use a sane default version, that will increase with the OS.
+
+Configuration options in this section can be provided the same way as they were in a cloud config, with the exception of dashes (`-`) being replaced with underscores (`_`) in key names.
+
+```container-linux-config:gce
+etcd:
+  name:                        "{HOSTNAME}"
+  advertise_client_urls:       "{PRIVATE_IPV4}:2379"
+  initial_advertise_peer_urls: "{PRIVATE_IPV4}:2380"
+  listen_client_urls:          "http://0.0.0.0:2379"
+  listen_peer_urls:            "http://{PRIVATE_IPV4}:2380"
+  initial_cluster:             "%m=http://{PRIVATE_IPV4}:2380"
+```
+
+### fleet
+
+Fleet is easily configurable in a cloud config, but due to the [planned deprecation of fleet][fleet-deprecation] support was not included in CLCs. Fleet can however still be configured with CLCs by use of a systemd drop-in.
+
+As an example, the following cloud config:
+
+```cloud-config
+#cloud-config
+
+coreos:
+  fleet:
+      public-ip: "$public_ipv4"
+      metadata: "region=us-west"
+```
+
+Could be converted into the following CLC.
+
+```container-linux-config
+systemd:
+  units:
+    - name: "fleet.service"
+      enable: true
+      dropins:
+        - name: "20-fleet-conf.conf"
+          contents: |
+            [Service]
+            Environment="FLEET_PUBLIC_IP=203.0.113.29"
```

Comment:
This doesn't actually use the public_ipv4 though. It might be easier to just use the static address in your example above.